shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -417,6 +417,50 @@ v8::MaybeLocal<v8::Value> PassValueToOtherContextInner(
       return v8::MaybeLocal<v8::Value>(
           blob.ToV8Value(destination_context->GetIsolate()));
     }
+
+    // Custom logic to "clone" FileList references
+    auto object_value = value.As<v8::Object>();
+    v8::Maybe<bool> has_length = object_value->Has(
```

Comment:
This is a fairly expensive and inaccurate check to run on every object, just because an object happens to match this structure we're gonna do things differently?

Reply:
i don't disagree - i just struggled to find a less magic approach to validating the object